services/jvm/src/main/scala/io/gearpump/services/MasterService.scala

```diff
@@ -188,6 +194,22 @@ class MasterService(val master: ActorRef,
       }
     }
   }
+
+  private def mergeConfig(configFile: Option[File], configString: String): Option[File] = {
+    if (configString == null || configString.isEmpty) {
+      configFile
+    } else {
+      configFile match {
+        case Some(file) =>
+          Files.write(file.toPath, ("\n" + configString).getBytes(UTF_8), APPEND)
```

Comment:
config string is also in HOCON.

And it works even if we directly combine HOCON + JSON.